src/executor/sort_agg.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2022 RisingLight Project Authors. Licensed under Apache-2.0.
+#![allow(dead_code)]
+use smallvec::SmallVec;
+
+use super::*;
+use crate::array::{ArrayBuilderImpl, ArrayImpl};
+use crate::binder::BoundAggCall;
+use crate::types::{DataTypeExt, DataTypeKind};
+
+pub struct SortAggExecutor {
+    pub agg_calls: Vec<BoundAggCall>,
+    pub group_keys: Vec<BoundExpr>,
+    pub child: BoxedExecutor,
+}
+
+impl SortAggExecutor {
+    #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
+    pub async fn execute(self) {
+        let mut last_key = None::<SmallVec<[DataValue; 16]>>;
+
+        #[for_await]
+        for chunk in self.child {
+            // Eval group keys and arguments
+            let chunk = chunk?;
+            let exprs: SmallVec<[ArrayImpl; 16]> = self
+                .agg_calls
+                .iter()
+                .map(|agg| agg.args[0].eval(&chunk))
+                .try_collect()?;
+            let group_cols: SmallVec<[ArrayImpl; 16]> = self
+                .group_keys
+                .iter()
+                .map(|e| e.eval(&chunk))
+                .try_collect()?;
+
+            let num_rows = chunk.cardinality();
+            let mut states = create_agg_states(&self.agg_calls);
```

Comment:
Should this variable be defined outside the `for chunk in self.child` loop?

Reply:
e.g., child may have two chunks `[1,1], [1,2]`

Reply:
You mean [1,1] [1,2], group key in index[0], the result is [3,2]？

Reply:
Yes